plugins/bufferlines/bufferline.nix

```
@@ -1,121 +1,192 @@
+        "enforceRegularTabs"
+        "alwaysShowBufferline"
+        "persistBufferSort"
+        "maxPrefixLength"
+        "sortBy"
+        "diagnostics"
+        "diagnosticsIndicator"
+        "diagnosticsUpdateInInsert"
+        "offsets"
+        [
+          "groups"
+          "items"
+        ]
+        [
+          "groups"
+          "options"
+          "toggleHiddenOnEnter"
+        ]
+        [
+          "hover"
+          "enabled"
+        ]
+        [
+          "hover"
+          "reveal"
+        ]
+        [
+          "hover"
+          "delay"
+        ]
+        [
+          "debug"
+          "logging"
+        ]
+        "customFilter"
+      ];
 
-      package = helpers.mkPluginPackageOption "bufferline" pkgs.vimPlugins.bufferline-nvim;
+      oldHighlightOptions = [
+        "fill"
+        "background"
+        "tab"
+        "tabSelected"
+        "tabSeparator"
+        "tabSeparatorSelected"
+        "tabClose"
+        "closeButton"
+        "closeButtonVisible"
+        "closeButtonSelected"
+        "bufferVisible"
+        "bufferSelected"
+        "numbers"
+        "numbersVisible"
+        "numbersSelected"
+        "diagnostic"
+        "diagnosticVisible"
+        "diagnosticSelected"
+        "hint"
+        "hintVisible"
+        "hintSelected"
+        "hintDiagnostic"
+        "hintDiagnosticVisible"
+        "hintDiagnosticSelected"
+        "info"
+        "infoVisible"
+        "infoSelected"
+        "infoDiagnostic"
+        "infoDiagnosticVisible"
+        "infoDiagnosticSelected"
+        "warning"
+        "warningVisible"
+        "warningSelected"
+        "warningDiagnostic"
+        "warningDiagnosticVisible"
+        "warningDiagnosticSelected"
+        "error"
+        "errorVisible"
+        "errorSelected"
+        "errorDiagnostic"
+        "errorDiagnosticVisible"
+        "errorDiagnosticSelected"
+        "modified"
+        "modifiedVisible"
+        "modifiedSelected"
+        "duplicate"
+        "duplicateVisible"
+        "duplicateSelected"
+        "separator"
+        "separatorVisible"
+        "separatorSelected"
+        "indicatorVisible"
+        "indicatorSelected"
+        "pick"
+        "pickVisible"
+        "pickSelected"
+        "offsetSeparator"
+      ];
 
-      mode = helpers.defaultNullOpts.mkEnumFirstDefault [
-        "buffers"
-        "tabs"
-      ] "mode";
+      basePluginPath = [
+        "plugins"
+        "bufferline"
+      ];
+      settingsPath = basePluginPath ++ [ "settings" ];
+      optionsPath = settingsPath ++ [ "options" ];
+      oldHighlightsPath = basePluginPath ++ [ "highlights" ];
+      newHighlightsPath = settingsPath ++ [ "highlights" ];
+    in
+    [ (lib.mkRenamedOptionModule (basePluginPath ++ [ "extraOptions" ]) optionsPath) ]
+    ++ (mkSettingsRenamedOptionModules basePluginPath optionsPath oldOptions)
+    ++ (mkSettingsRenamedOptionModules oldHighlightsPath newHighlightsPath oldHighlightOptions)
```

I don't mind if you think they add clarity, but these parenthesis are redundant. (They would be needed if the expressions were list items, since the list syntax uses whitespace as a deliminator)

```suggestion
    ++ mkSettingsRenamedOptionModules basePluginPath optionsPath oldOptions
    ++ mkSettingsRenamedOptionModules oldHighlightsPath newHighlightsPath oldHighlightOptions
```

---

Also, should we consider having the `trunc_marker` and `extraOptions` in the same `[ ]` list, so that we don't have multiple list literals in the concatenation?

---

Yeah, I can try that. I think there was an iteration during development where the order of execution did matter.